js/ui/appDisplay.js

```diff
@@ -2743,3 +2771,64 @@ class AppCenterIcon extends AppIcon {
         return true;
     }
 });
+
+const HackAppIconState = {
+    ACTIVATED: ViewIconState.NUM_STATES + 2,
+    DEACTIVATED: ViewIconState.NUM_STATES + 3
+};
+
+var HackAppIcon = GObject.registerClass(
+class HackAppIcon extends ViewIcon {
+    _init(parentView) {
+        let viewIconParams = { isDraggable: false,
+                               showMenu: false,
+                               parentView: parentView };
+
+        let buttonParams = { button_mask: St.ButtonMask.ONE | St.ButtonMask.TWO };
+        let iconParams = { createIcon: this._createIcon.bind(this) };
+
+        super._init(viewIconParams, buttonParams, iconParams);
+
+        global.settings.connect('changed::hack-mode-enabled', () => {
+            let activated = global.settings.get_boolean('hack-mode-enabled');
+            this.iconState = activated ? HackAppIconState.ACTIVATED : HackAppIconState.DEACTIVATED;
+        });
+
+        this.canDrop = false;
+
+        this._iconContainer = new St.Widget({ layout_manager: new Clutter.BinLayout(),
+                                              x_expand: true, y_expand: true });
+        this._iconContainer.add_child(this.icon.actor);
+
+        this.actor.set_child(this._iconContainer);
+
+        this.actor.connect('destroy', this._onDestroy.bind(this));
+    }
+
+    _createIcon(iconSize) {
+        let iconUri = 'resource:///org/gnome/shell/theme/hack-button.svg';
+        if (this._iconState === HackAppIconState.DEACTIVATED)
```

Comment:
`this.iconState` instead of `this._iconState`

Reply:
It's the same, because the `iconState` is a getter for the `_iconState`, but yes, we should use the getter instead the attribute itself, I'll change that.